### PR TITLE
Add Notifications#NewComment worker

### PR DESCRIPTION
**Pull request by bunufi**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

Added Notifications#NewComment worker. Will create a separate PR for removing the old job. 

## Related Tickets & Documents

#5305

## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [x] no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

je

### CODE REVIEW

File: app/models/notification.rb [hunk 1]
```diff
@@ -47,14 +47,14 @@ def send_new_comment_notifications(comment)
       return if comment.commentable_type == "PodcastEpisode"
```
Comment (bunufi) [inline app/models/notification.rb]: On a side note. I was not able to find where this method is being called - **send_new_comment_notifications**
Could it be that it is not needed and is better deleted?

**Comment by bunufi** [comment]
CodeClimate is failing, but not exactly sure what I can do with that. Please advise. :)

File: app/models/notification.rb (hunk 1 above)
Comment (mstruve) [reply app/models/notification.rb]: Yeah, I did a search myself and could not find it used anywhere. For the sake of simplicity, I say lets delete it.

File: app/workers/notifications/new_comment_worker.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+module Notifications
+  class NewCommentWorker
+    include Sidekiq::Worker
```
Comment (mstruve) [inline app/workers/notifications/new_comment_worker.rb]: [CodeClimate](https://codeclimate.com/github/thepracticaldev/dev.to/pull/5565) is saying that this and the other Notification workers have similar blocks of code and that we should refactor them. I think you are fine to ignore it.

Comment (rhymes) [reply app/workers/notifications/new_comment_worker.rb]: yeah, it's fine

File: app/models/notification.rb (hunk 1 above)
Comment (bunufi) [reply app/models/notification.rb]: Added a commit to this PR that removes the method discussed.

**Comment by bunufi** [comment]
Is there any other way I can indicate this PR is ready for a review other than ping? :) @mstruve @rhymes

**Comment by bunufi** [comment]
Hey @rhymes, If I understand correctly, if we move the `Notifications::NewComment::Send` service call out of the background job, it would prolong the http request of posting comments by the amount of time it takes to run the service. 

At the moment I feel that it should stick in the background job for performance reasons, but I lack a better understanding of the app. Maybe @mstruve could help answering this. :)

**Comment by rhymes** [comment]
@bunufi good point, but that's what we're already doing

The method is called `send_new_comment_notifications_without_delay` exactly because it's run inline and not delayed (a legacy nomenclature we had in the code base for jobs run in the background).

The body of `send_new_comment_notifications_without_delay` calls `      Notifications::NewCommentWorker.new.perform(comment.id)` which *does not* enqueue a job in the background. It runs it's content inline, like a normal class + method call.

That's why I was suggesting removing the job altogether and just calling the service.

Let's wait for @mstruve's feedback on this before adding additional work to this PR

Thanks @bunufi!

**Comment by mstruve** [comment]
Good catch @rhymes. @bunufi I think @rhymes makes a very good point that if we are not going to background this work at all there is no reason to have a worker then. Instead, we end up with this in the `send_new_comment_notifications_without_delay` method
```ruby
    def send_new_comment_notifications_without_delay(comment)
      return if comment.commentable_type == "PodcastEpisode"
      return if UserBlock.blocking?(comment.commentable.user_id, comment.user_id)

      Notifications::NewComment::Send.call(comment)
    end
``` 

Not only do we save ourselves the complexity of an additional worker and specs in the codebase but we also eliminate looking up the comment again using the ID. Let me know if that makes sense or if you have any questions!!!

**Comment by bunufi** [comment]
I am so sorry for not knowing what `Notifications::NewCommentWorker.new.perform(comment.id)` does. :) Since it said perform and was named worker I have blindly accepted the fact that it must act like a background job. Now I know! 

I have made the change, removed the worker and its spec. 

Thank you so much for your patience with me @rhymes and @mstruve!

**Comment by rhymes** [comment]
Awesome @bunufi, thank you and sorry for not catching this sooner!

**Comment by bunufi** [comment]
You are all too nice here @rhymes and @mstruve . I am not used to this. :) 
It was a great lesson for me, next time it should not slip though my eyes either. Thanks for patience and explanations!

**Pull request merged**